src/plugins/Network.js

```diff
@@ -47,26 +60,69 @@ module.exports = class Network extends OverlayPlugin {
          *            down_pkt_loss:<enabled/disabled>:<value>
          *            dns_delay:<enabled/disabled>:<value>
          */
-        this.instance.registerEventCallback('network_profile', (message) => {
-            const values = message.split(' ');
-            if (values.length < 9) {
+        this.instance.registerEventCallback('network_profile', this.handleNetworkProfile.bind(this));
+    }
+
+    // Handle settings event to enable/disable wifi|mobile data
+    handleSettings(message) {
+        const values = message.split(' ');
+
+        if (values[0] === 'if') {
+            this.wifiInput.disabled = false;
+            this.mobileInput.disabled = false;
+
+            if (values.length !== 3) {
                 return;
             }
-            const upSpeed = values[2].split(':');
-            const downSpeed = values[3].split(':');
-            const upDelay = values[4].split(':');
-            const downDelay = values[5].split(':');
-            const upPacketLoss = values[6].split(':');
-            const downPacketLoss = values[7].split(':');
-            const dnsDelay = values[8].split(':');
-
-            const isThrottlingEnabled = upSpeed[1] === 'enabled'
-                && downSpeed[1] === 'enabled'
-                && upDelay[1] === 'enabled'
-                && downDelay[1] === 'enabled'
-                && upPacketLoss[1] === 'enabled'
-                && downPacketLoss[1] === 'enabled'
-                && dnsDelay[1] === 'enabled';
+
+            const wifiOn = values[1].match(/(wifi:)(\w+)/);
+            if (wifiOn) {
+                this.wifiInputChecked = wifiOn[2] === 'on';
+            }
+            const mobileOn = values[2].match(/(mobile:)(\w+)/);
+            if (mobileOn) {
+                this.mobileInputChecked = mobileOn[2] === 'on';
+            }
+
+            if (this.widgetRendered) {
+                this.wifiInput.checked = this.wifiInputChecked;
+                this.mobileInput.checked = this.mobileInputChecked;
+                this.updateMobileSectionStatus();
+            }
+        } else if (values[0] === 'parameter' && values.length > 2 && values[2].includes('android_version')) {
+            const version = values[2].match(/(android_version:)(\w+)/);
+            this.androidVersion = version[2];
```

Comment:
What are the possible values for `version`? Can we have `8.0` and `8.1` for oreo? or is it just the major version (`8`) which is stored?

I'm wondering because it looks like `this.androidVersion` may be a string, but we compare it to ints (the constants `ANDROID_8` and `ANDROID_10`.  I have a doubt as to how a string `"11"` would compare to an int `8`.

Reply:
JavaScript allow comparison between string and number :) 
https://stackoverflow.com/a/28994875/2528699

Reply:
I agree with Carmen here: I know it's possible on the technical side to compare strings & numbers, but the issue I see here is that we're specifically comparing version numbers ?

I just tested a bit in the console some variations of `"9" <= 8.1` (moving around the type and values). It seems to give good results as long as we stay under 10, since at that point alphabetic ordering can get different from numeric ordering.

It feels a bit risky to me. At the very least, forcing the conversion to number could be a good thing IMO.

Reply:
![image](https://user-images.githubusercontent.com/69793750/208479147-80d27549-88ef-4216-a87e-4d5b192b2946.png)

Reply:
I don't see any more version comparisons :+1: This point looks resolved to me!